internal/aws/awsutil/conn.go

```diff
@@ -226,119 +206,40 @@ func ProxyServerTransport(logger *zap.Logger, config *AWSSessionSettings) (*http
 	return transport, nil
 }
 
-func (c *Conn) newAWSSession(logger *zap.Logger, roleArn, externalID string, region string) (*session.Session, error) {
-	var s *session.Session
-	var err error
-	if roleArn == "" {
-		s, err = GetDefaultSession(logger)
-		if err != nil {
-			return s, err
-		}
-	} else {
-		stsCreds, _ := getSTSCreds(logger, region, roleArn, externalID)
-
-		s, err = session.NewSession(&aws.Config{
-			Credentials: stsCreds,
-		})
-		if err != nil {
-			logger.Error("Error in creating session object : ", zap.Error(err))
-			return s, err
-		}
+// getPartition returns AWS Partition for the provided region.
+func getPartition(region string) string {
+	// This is a simplified version - in production we would use the actual AWS SDK v2
```

Comment:
We should definitely use the actual AWS SDK partitions package

Reply:
I will check if there are any other parts that I missed, not just that part. Thank you for the review.

Reply:
@mjlshen  AWS SDK v2 provides the GetPartition function in the internal module so, i worked in reference(https://github.com/aws/aws-sdk-go-v2/blob/main/internal/endpoints/awsrulesfn/partitions.go)